## Calculating Covariance and Correlation

Consider a portfolio composed of *Walmart* and *Facebook*. Do you expect the returns of these companies to show high or low covariance? Or, could you guess what the correlation would be? Will it be closer to 0 or closer to 1? 

Since Walmart and Facebook do not exist in the same business vertical, my intuition tells me that they will be weakly correlated, close to 0. 

Begin by extracting data for Walmart and Facebook from the 1st of January 2014 until today.

In [1]:
import numpy as np
import pandas as pd
from pandas_datareader import data as wb
import matplotlib.pyplot as plt

In [3]:
sec_data = pd.read_csv('Walmart_FB_2014_2017.csv', index_col='Date')
sec_data.tail()

WMT          FB
Date                             
2017-04-03  70.906502  142.279999
2017-04-04  71.084190  141.729996
2017-04-05  70.728821  141.850006
2017-04-06  70.511650  141.169998
2017-04-07  71.962746  140.779999

In [4]:
sec_returns = np.log(sec_data / sec_data.shift(1))

Repeat the process we went through in the lecture for these two stocks. How would you explain the difference between their means and their standard deviations?

### Walmart

In [5]:
wmt_avg_annual_returns = sec_returns['WMT'].mean() * 250
wmt_annual_std = sec_returns['WMT'].std() * 250 ** 0.5

print(wmt_avg_annual_returns)
print(wmt_annual_std)

0.0035509835603423465
0.177606600224821


### Facebook

In [6]:
fb_avg_annual_returns = sec_returns['FB'].mean() * 250
fb_annual_std = sec_returns['FB'].std() * 250 ** 0.5

print(fb_avg_annual_returns)
print(fb_annual_std)

0.2874389823382856
0.28772593827402126


It looks like WMT has on average near 0 return with 17.76% standard deviation. While facebook enjoys a generate 28.7% with 28.77% standard deviation, which is much more volatile. 

***

## Covariance and Correlation


\begin{eqnarray*}
Covariance Matrix: \  \   
\Sigma = \begin{bmatrix}
        \sigma_{1}^2 \ \sigma_{12} \ \dots \ \sigma_{1I} \\
        \sigma_{21} \ \sigma_{2}^2 \ \dots \ \sigma_{2I} \\
        \vdots \ \vdots \ \ddots \ \vdots \\
        \sigma_{I1} \ \sigma_{I2} \ \dots \ \sigma_{I}^2
    \end{bmatrix}
\end{eqnarray*}

Covariance matrix:

In [8]:
cov_matrix = sec_returns.cov() * 250
cov_matrix

WMT        FB
WMT  0.031544  0.008777
FB   0.008777  0.082786

Correlation matrix:

In [9]:
corr_matrix = sec_returns.corr()
corr_matrix

WMT        FB
WMT  1.000000  0.171746
FB   0.171746  1.000000

Would you consider investing in such a portfolio?